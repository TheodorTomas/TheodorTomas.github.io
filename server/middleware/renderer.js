import React from 'react';
import { renderToString } from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import App from '../../src/containers/App'; // import our main App component
const path = require('path');
const fs = require('fs');
export default (req, res) => {
  // point to the html file created by CRA's build tool
  const filePath = path.resolve(__dirname, '..', '..', 'build', 'index.html');
  fs.readFile(filePath, 'utf8', (err, htmlData) => {
    if (err) {
      console.error('err', err); // eslint-disable-line
      return res.status(404).end();
    }
    const sheet = new ServerStyleSheet();
    // render the app as a string
    const html = renderToString(sheet.collectStyles(<App />));
    const css = sheet.getStyleTags();
    // inject the rendered app into our html and send it
    return res.send(
      htmlData
        .replace(
          '<style ssr-styles></style>',
          css
        )
        .replace(
          '<div id="root"></div>',
          `<div id="root">${html}</div>`
        )
    );
  });
};
