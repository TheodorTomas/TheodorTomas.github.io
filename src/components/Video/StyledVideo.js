import styled from 'styled-components';
import { COLORS } from '../../containers/App/theme';

export const StyledVideo = styled.video`
  /* Sizing */
  width: 150px;
  height: 150px;
  margin: auto;
  
  /* Misc */
  background-color: ${COLORS.videoBgColor.hex};
  overflow: hidden;
  transition: border-color 0.25s ease-in-out, transform 0.25s ease-in-out;
  border-radius: 50%;
  border: 2px solid;
  border-color: ${COLORS.darkGrey.rgba(0.5)};
  box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.2);
  
  ${({ isClickable }) => (
    isClickable
      ? `
        &:hover {
          transform: scale(1.01);
          cursor: pointer;
          border-color: ${COLORS.blueGrey.rgba(0.5)};
        }
        `
      : ''
  )}
`;
