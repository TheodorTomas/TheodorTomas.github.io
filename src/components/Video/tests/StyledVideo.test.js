import React from 'react';
import renderer from 'react-test-renderer';
import { StyledVideo } from '../StyledVideo';

describe('render()', () => {
  it('should render component matching the snapshot when isClickable = false', () => {
    const component = renderer.create(<StyledVideo isClickable={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component matching the snapshot when isClickable = true', () => {
    const component = renderer.create(<StyledVideo isClickable />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
