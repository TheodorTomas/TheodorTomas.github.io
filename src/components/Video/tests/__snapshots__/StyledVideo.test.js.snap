// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() should render component matching the snapshot when isClickable = false 1`] = `
.c0 {
  width: 150px;
  height: 150px;
  margin: auto;
  background-color: #8f8c86;
  overflow: hidden;
  -webkit-transition: border-color 0.25s ease-in-out,-webkit-transform 0.25s ease-in-out;
  -webkit-transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  border-radius: 50%;
  border: 2px solid;
  border-color: rgba(74,74,74,0.5);
  box-shadow: 2px 2px 2px 1px rgba(0,0,0,0.2);
}

<video
  className="c0"
/>
`;

exports[`render() should render component matching the snapshot when isClickable = true 1`] = `
.c0 {
  width: 150px;
  height: 150px;
  margin: auto;
  background-color: #8f8c86;
  overflow: hidden;
  -webkit-transition: border-color 0.25s ease-in-out,-webkit-transform 0.25s ease-in-out;
  -webkit-transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  border-radius: 50%;
  border: 2px solid;
  border-color: rgba(74,74,74,0.5);
  box-shadow: 2px 2px 2px 1px rgba(0,0,0,0.2);
}

.c0:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  cursor: pointer;
  border-color: rgba(149,163,179,0.5);
}

<video
  className="c0"
/>
`;
